go/worker/compute/executor/committee/node.go

```diff
@@ -724,25 +724,49 @@ func (n *Node) proposeTimeoutLocked() error {
 }
 
 func (n *Node) handleScheduleBatch(force bool) {
-	epoch, lastHeader, err := func() (*committee.EpochSnapshot, *block.Header, error) {
+	epoch, lastHeader, roundResults, err := func() (*committee.EpochSnapshot, *block.Header, *roothash.RoundResults, error) {
 		n.commonNode.CrossNode.Lock()
 		defer n.commonNode.CrossNode.Unlock()
 
 		// If we are not waiting for a batch, don't do anything.
 		if _, ok := n.state.(StateWaitingForBatch); !ok {
-			return nil, nil, errIncorrectState
+			return nil, nil, nil, errIncorrectState
 		}
 		if n.commonNode.CurrentBlock == nil {
-			return nil, nil, errNoBlocks
+			return nil, nil, nil, errNoBlocks
 		}
 		header := n.commonNode.CurrentBlock.Header
 		epoch := n.commonNode.Group.GetEpochSnapshot()
 
 		// If we are not an executor worker in this epoch, we don't need to do anything.
 		if !epoch.IsExecutorWorker() {
-			return nil, nil, errNotTxnScheduler
+			return nil, nil, nil, errNotTxnScheduler
 		}
-		return epoch, &header, nil
+
+		// Fetch message results emitted during the last normal round.
```

Comment:
This should probably be factored out so it isn't duplicated in `startProcessingBatchLocked`.

Reply:
done